source/extensions/filters/http/on_demand/on_demand_update.cc

```diff
@@ -10,16 +11,20 @@ namespace HttpFilters {
 namespace OnDemand {
 
 Http::FilterHeadersStatus OnDemandRouteUpdate::decodeHeaders(Http::RequestHeaderMap&, bool) {
-  if (callbacks_->route() != nullptr ||
-      !(callbacks_->routeConfig().has_value() && callbacks_->routeConfig().value()->usesVhds())) {
+
+  if (callbacks_->route() != nullptr) {
     filter_iteration_state_ = Http::FilterHeadersStatus::Continue;
     return filter_iteration_state_;
   }
+  // decodeHeaders() is interrupted.
+  decode_header_interrupted_ = true;
```

Comment:
What do you mean by "interrupted"? I think what is actually being tracked here is "are we inside decodeHeaders() processing?", so `decode_headers_active_` might make more sense.

Reply:
Yes, that make sense.